### ISSUE TITLE
Add support for the new architecture

### ISSUE THREAD

**Issue by capezzbr** [react-native-datetimepicker/datetimepicker#622]
Was wondering if you had plans to add support for the new Fabric renderer

**Comment by vonovak** [comment]
Hello,
Thanks for asking. Yes, the new renderer will be supported but I cannot give a specific time line.
Thank you 🙂

**Comment by capezzbr** [comment]
@vonovak thanks for following up! I know it's hard to provide a timeline, but it believe this would help folks relying on this library that want to enable Fabric 🙇

**Comment by tom-sherman** [comment]
Will this work also include turbomodule support? If so, can we change the issue title?

**Comment by BrunoVillanova** [comment]
Hey guys,

I can help with that.
What is the current status of this issue?
Has anyone already started?

**Comment by alfonsocj** [comment]
Hey, I just started today on supporting Fabric for iOS.

**Comment by BrunoVillanova** [comment]
Awesome!
Please tell me if you need some help.
☺️

**Comment by capezzbr** [comment]
PR in review https://github.com/react-native-datetimepicker/datetimepicker/pull/657

**Comment by vonovak** [comment]
reopening since android is still pending, so I'll reopen

**Comment by alfonsocj** [comment]
I think the example is the one not compatible. I'll investigate further.

**Comment by ShrutiRani7029** [comment]
HEY, I HAVE DONE MY PROJECTS ON REACT NATIVE WHERE I INNOVATE AND ADD NEW FEATURES, CAN YOU GIVE ME A CHANCE SO THAT I ADD A NEW ARCHITECTURE?

**Comment by EkaterinaMozheiko** [comment]
Is there any update about implementing Fabric support for Android?

**Comment by HarshaR1642** [comment]
> Is there any update about implementing Fabric support for Android?

+1

**Comment by vonovak** [comment]
Hello, the android part of this module does not really render anything in react native, and therefore android will _not_ need to support Fabric because it will never use Fabric. However, we will need to support the new turbomodules.
I'm planning to add turbomodules support to another module I maintain and then add the support here. My timeline is that I'll prob start in ~1.5 months. If anyone wants to do that before, please let us know here and go ahead! 👍 @ShrutiRani7029

**Comment by capezzbr** [comment]
Also, if you are experiencing an issue building your Android app with the new architecture enabled on RN 0.70, you should add this snippet of code to your `react-native.config.js`

```json
module.exports = {
  dependencies: {
    ...
    // https://github.com/react-native-community/cli/blob/0e63e750a235062cd9bc43ed6a4a2beb8f14385a/docs/autolinking.md#how-can-i-disable-autolinking-for-new-architecture-fabric-turbomodules
    '@react-native-community/datetimepicker': {
      platforms: {
        android: {
          libraryName: null,
          componentDescriptors: null,
          androidMkPath: null,
          cmakeListsPath: null,
        },
      },
    },
  },
```

the issue from my understanding is that `@react-native-community/datetimepicker` doesn't have any Android UI components that needs to be autolinked, but RN CLI does detect what is defined in the module `package.json`, which are only for iOS. I believe a better workaround would be to created a `react-native.config.js` in the `@react-native-community/datetimepicker`, that does the above. I am happy to open a PR with this fix soon

**Comment by HarshaR1642** [comment]
> Also, if you are experiencing an issue building your Android app with the new architecture enabled on RN 0.70, you should add this snippet of code to your `react-native.config.js`
> 
> ```
> module.exports = {
>   dependencies: {
>     ...
>     // https://github.com/react-native-community/cli/blob/0e63e750a235062cd9bc43ed6a4a2beb8f14385a/docs/autolinking.md#how-can-i-disable-autolinking-for-new-architecture-fabric-turbomodules
>     '@react-native-community/datetimepicker': {
>       platforms: {
>         android: {
>           libraryName: null,
>           componentDescriptors: null,
>           androidMkPath: null,
>           cmakeListsPath: null,
>         },
>       },
>     },
>   },
> ```
> 
> the issue from my understanding is that `@react-native-community/datetimepicker` doesn't have any Android UI components that needs to be autolinked, but RN CLI does detect what is defined in the module `package.json`, which are only for iOS. I believe a better workaround would be to created a `react-native.config.js` in the `@react-native-community/datetimepicker`, that does the above. I am happy to open a PR with this fix soon

Hi @capezzbr 
I added your temporary fix and tried but it is not working, still facing the same issue with android

**Comment by capezzbr** [comment]
@HarshaR1642 what exactly is the error you are experiencing

**Comment by Josh2941** [comment]
DatePicker is not rendering on Android. On iOS Simulator it works fine. I am on RN 0.70.6.

**Comment by BraveEvidence** [comment]
f interested I have achieved this functionality for new architecture using fabric https://www.youtube.com/watch?v=W3m-8_QuC14&list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i&index=4 and
https://www.youtube.com/watch?v=38dpw_s2YKE&list=PLQhQEGkwKZUrempLnmxjt7ZCZJu1W3p2i&index=3

**Comment by vonovak** [comment]
The android support is happening here: https://github.com/react-native-datetimepicker/datetimepicker/pull/730

**Comment by vonovak** [comment]
closed via #730 🎉
